lib/que/active_job/extensions.rb

```diff
@@ -82,11 +82,6 @@ def run(args)
           # option under the circumstances (doesn't require hacking ActiveJob in
           # any more extensive way).
 
-          # There's no reason this logic should ever nest, because it wouldn't
-          # make sense to run a worker inside of a job, but even so, assert that
-          # nothing absurd is going on.
-          Que.assert NilClass, Thread.current[:que_current_job]
-
```

Comment:
This method has only been overridden by an ActiveJob test support helper specifically to prevent this assertion from making the Rails test suite fail. This assertion still runs in production on Rails.

It's possible that removing the assertion is ok, but I would like to have a better grasp as to why it was added in the first place before we do. If there is some logic downstream from this point that depends on the fact that jobs are not nested, then removing the assertion may make the Rails test suite pass but could break que.